src/main/java/frc/robot/component/Climber.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.component;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkPIDController;
+import edu.wpi.first.units.Current;
+import edu.wpi.first.units.Time;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import frc.robot.util.MoPrefs;
+import frc.robot.util.MoPrefs.UnitPref;
+import frc.robot.util.MoShuffleboard;
+import frc.robot.util.MoUnits;
+
+/* A single side climber */
+public class Climber {
+    public final CANSparkMax winch;
+    public final SparkPIDController winchPID;
+    public final CurrentTrigger zeroLimit;
+    private boolean hasZero = false;
+
+    public Climber(
+            String name, CANSparkMax winch, UnitPref<Current> zeroCurrentLimit, UnitPref<Time> zeroTriggerDuration) {
+        this.winch = winch;
+        this.winchPID = winch.getPIDController();
+        this.zeroLimit = CurrentTrigger.ofSparkMax(winch, zeroCurrentLimit, zeroTriggerDuration);
+
+        var layout = MoShuffleboard.getInstance()
+                .climberTab
+                .getLayout(name, BuiltInLayouts.kList)
+                .withSize(1, 2);
+        layout.addBoolean("At Zero Limit", this::atZeroLimit);
+        layout.addBoolean("Has Zero", this::hasZero);
+        layout.addDouble("Encoder Pos", () -> this.winch.getEncoder().getPosition());
+        layout.addDouble("Current", () -> this.winch.getOutputCurrent());
+    }
+
+    public void invalidateZero() {
+        hasZero = false;
+    }
+
+    public boolean hasZero() {
+        return hasZero;
+    }
+
+    private void runWinch(double power) {
+        if (MoPrefs.climberPid.get()) {
+            winchPID.setReference(
+                    power * MoPrefs.climberMotorSpeed.get().in(MoUnits.RotationsPerMinute),
+                    CANSparkMax.ControlType.kVelocity);
```

Comment:
Hmm, we shouldn't be using raw SparkPIDControllers either. This should be wrapped in a MoSparkPIDController, which provides two benefits:
1. It allows the use of the TunerUtils to tune the PID constants.
2. It handles the units. So you just pass your Velocity measure in, and it will determine the units necessary to be sent to the PID controller.

Refer to the ArmSubsystem/IntakeSubsystem/SwerveModule classes for examples.

Reply:
Although actually we should just take the PID out entirely. It's not necessary in this situation.